### PR TITLE
add `frontier/ethereum` example

### PR DESCRIPTION
**Pull request by niklasad1**
The ethereum config doesn't work with the default PolkadotConfig which this shows how to do.

Tested with a frontier node

### CODE REVIEW

File: subxt/examples/frontier_eth.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    type AssetId = u32;
+}
+
+// This helper makes it easy to use our `eth_signer::AccountId20`'s with generated
+// code that expects a generated `eth_runtime::runtime_types::foo::AccountId20` type.
+// an alternative is to do some type substitution in the generated code itself, but
+// mostly I'd avoid doing that unless absolutely necessary.
+fn to_runtime_account_id(a: AccountId20) -> eth_runtime::runtime_types::fp_account::AccountId20 {
+    eth_runtime::runtime_types::fp_account::AccountId20(a.0)
+}
```
Comment (jsdw) [inline subxt/examples/frontier_eth.rs:29]: Could also `impl From`, but for the sake of the example this is easy to follow :)

File: subxt/examples/frontier_eth.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#![allow(missing_docs)]
+
+use subxt::OnlineClient;
+use subxt_signer::eth::{dev, AccountId20, Signature};
+
+#[subxt::subxt(runtime_metadata_insecure_url = "ws://127.0.0.1:9944")]
+mod eth_runtime {}
```
Comment (jsdw) [inline subxt/examples/frontier_eth.rs:7]: prob worth a comment that in order to run this example, somebody should be running a frontier node locally (and maybe the comment can briefly say how eg `frontier --dev` or whatever it is).

File: subxt/examples/frontier_eth.rs
Comment (jsdw) [inline subxt/examples/frontier_eth.rs:1]: In general, I think examples should be linked in the book somewhere.

For that reason, could we also follow the orugh naming convention and have it called something like `tx_basic_frontier.rs` maybe?

**Review by jsdw — APPROVED** [review]
Awesome!

I could have sworn that we had a very similar example somewhere, but I couldn't find it offhand so this looks good  to me!

File: subxt/examples/tx_basic_frontier.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let api = OnlineClient::<EthRuntimeConfig>::from_insecure_url("ws://127.0.0.1:9944").await?;
+
+    let alith = dev::alith();
+    let baltathar = dev::baltathar();
+    let dest = baltathar.account_id();
+
+    println!("baltathar pub:  {}", hex::encode(&baltathar.public_key().0));
```
Comment (niklasad1) [inline subxt/examples/tx_basic_frontier.rs:44]:
```suggestion
    println!("baltathar pub:  {}", hex::encode(baltathar.public_key().0));
```

File: subxt/examples/tx_basic_frontier.rs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let api = OnlineClient::<EthRuntimeConfig>::from_insecure_url("ws://127.0.0.1:9944").await?;
+
+    let alith = dev::alith();
+    let baltathar = dev::baltathar();
+    let dest = baltathar.account_id();
+
+    println!("baltathar pub:  {}", hex::encode(baltathar.public_key().0));
+    println!("baltathar addr: {}", hex::encode(&dest));
```
Comment (niklasad1) [inline subxt/examples/tx_basic_frontier.rs:45]:
```suggestion
    println!("baltathar addr: {}", hex::encode(dest));
```

**Comment by niklasad1** [comment]
@CodiumAI-Agent /improve

**Comment by niklasad1** [comment]
@CodiumAI-Agent /describe

**Pull request merged**